public/usergameactivity.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+use RA\AchievementType;
+use RA\Permissions;
+
+if (!authenticateFromCookie($user, $permissions, $userDetails, Permissions::Admin)) {
+    abort(401);
+}
+
+$gameID = requestInputSanitized('ID', null, 'integer');
+$user2 = requestInputSanitized('f');
```

Comment:
I'd probably go with `$targetUser`, or `$targetUsername` even, instead of `$user2`.

The parameter can be `u` instead of `f` as it's not used here and seems more fitting?

```php
$gameID = (int) request()->input('ID');
$targetUser = request()->input('u');
```

Reply:
Heads up: eventually the route might be mapped to something like `user/{username}/game/{gameId}/activity`. Improving the routing & redirecting legacy URLs to pretty URLs is on my list for after the mysqli refactoring.

Reply:
That was just copy/pasted from gamecompare.php:
https://github.com/RetroAchievements/RAWeb/blob/564e3bc7b1eeb1407a234a401894b4bed936bad9/public/gamecompare.php#L9-L10

Reply:
That's what I assumed :) Can stay as is imo.